app/coffeescripts/views/conferences/ConferenceView.coffee

```diff
@@ -126,3 +128,83 @@ define [
         else
           window.open(data[0].url)
       )
+
+    deleteRecording: (e) ->
+      e.preventDefault()
+      if !confirm I18n.t("Are you sure you want to delete this recording?")
+        $(e.currentTarget).parents('.inline-block').find('.al-trigger').focus()
+        return
+      $deleteButton = $(e.currentTarget).parent()
+      @toggleActionButton($deleteButton, {state: "processing", action: "delete"})
+      @toggleRecordingLink($deleteButton, {state: "processing"})
+      $.ajaxJSON($deleteButton.data('url') + "/recording", "DELETE", {
+          recording_id: $deleteButton.data("id"),
+        }, (data) =>
+          if data.deleted == "true"
+            @removeRecordingRow($deleteButton)
+            return
+          @ensureActionPerformed($deleteButton, 1, @removeRecordingRow)
+      )
+
+    ensureActionPerformed: ($button, attempt, callback) =>
+      $.ajaxJSON($button.data('url') + "/recording", "GET", {
+          recording_id: $button.data("id"),
+        }, (data) =>
+          if $.isEmptyObject(data)
+            callback($button)
+            return
+          if attempt <= 5
+            attempt++
+            setTimeout((=> @ensureActionPerformed($button, attempt, callback); return;), attempt * 1000)
+            return
+          $.flashError(I18n.t("Sorry, the action performed on this recording failed. Try again later"))
+          @toggleActionButton($button, {state: "processed", action: "delete"})
+          @toggleRecordingLink($button, {state: "processed"})
+      )
+
+    toggleActionButton: ($button, data) =>
+      $spinner = $('.ig-loader[data-id="' + $button.data("id") + '"][data-action="' + data.action + '"]')
+      if data.state == 'processing'
+        $button.hide()
+        $spinner.show()
+        return
+      $spinner.hide()
+      $button.show()
+
+    toggleRecordingLink: ($button, data) =>
+      $link = $('a[data-id="' + $button.data("id") + '"]')
+      if data.state == 'processing'
+          $link.bind 'click', ->
+            return false
+          return
+      $link.unbind 'click'
+
+    removeRecordingRow: ($button) =>
+      $row = $('.ig-row[data-id="' + $button.data("id") + '"]')
+      containerId = $($row.parent().parent().parent()).attr('id')
+      $row.parent().remove()
+      id = containerId.substring(11, containerId.length)
+      @updateConferenceDetails(containerId.substring(11, containerId.length))
+      $.screenReaderFlashMessage(I18n.t('Recording was deleted'))
+
+    updateConferenceDetails: (id) =>
+      $info = $('div.ig-row#conf_' + id).children().children('div.ig-info')
+      $detailRecordings = $info.children('div.ig-details').children('div.ig-details__item-recordings')
+      $recordings = $('.ig-sublist#conference-' + id)
+      recordings = $recordings.children().children().length
+      # If it has more than one recording
```

Comment:
The code below is self documenting enough we don't need this comment

Reply:
Done